### ISSUE TITLE
FAR Flaps + Spoilers

### ISSUE THREAD

**Issue by DeputyLOL** [Mihara/RasterPropMonitor#280]
Not sure if this is where to put it.
I installed FAR for its flaps and spoilers usage to add a bit more realism (and headaches as most my planes need redesigning. 
In FAR there are 4 flap settings:
0 - Fully Retracted
1 - For climb
2 - For take off
3 - For landing
Is it possible to have the flaps position as a variable in RPM?

**Comment by Alexustas** [comment]
I'm fully support this request, also it would be great to have an appropriate Action handler to control the flaps

**Comment by MOARdV** [comment]
I'll look into support FAR flap settings and an action handler.

**Comment by MOARdV** [comment]
It doesn't look like FAR is set up to allow me to do this just yet, so I've filed a feature request to expose the information and functionality.

**Comment by MOARdV** [comment]
ferram is amenable to exposing the variables and control needed for RPM to relay FAR flap / spoiler data.  It will require an updated version of FAR, so once that's available, I'll integrate these features.

**Comment by DeputyLOL** [comment]
Thanks man, cant wait for v0.21! Any idea when it will be finished?

**Comment by MOARdV** [comment]
I'm waiting on MechJeb to release an official update.  Sarbian made changes in MJ after 2.5.1 that I have to account for.  I don't want to release RPM officially against a MJ dev build, so I'm on hold right now.  Which is why I keep slipping more stuff into v0.21.0

**Comment by DeputyLOL** [comment]
fair enough. I see the HUD ladder issue is still a problem though.

**Comment by MOARdV** [comment]
Yes, the ladder issue is still present.  If things quiet down, I'll spend some time on it to see if I can get that sorted out.  I've got a bug open for that issue.

**Comment by MOARdV** [comment]
It looks like FAR support is ready, as of the comments in over [here](https://github.com/ferram4/Ferram-Aerospace-Research/issues/96).

**Comment by MOARdV** [comment]
Will an IncreaseFlaps and DecreaseFlaps function be sufficient, or does there need to be a way to set flaps to a specific value (like SetFlaps(0) or SetFlaps(2) or whatever)?

**Comment by Alexustas** [comment]
I think  IncreaseFlaps / DecreaseFlaps will be enough

**Comment by MOARdV** [comment]
I have an initial dev build ready.  I've semi-tested it, but I don't fly spaceplanes, so most of these features are outside my normal style of gameplay.  For some reason, GetFlapSetting always returns 2 for me, even when I see the flaps are in a different position from outside the craft (when I right click on the flaps).  I don't know if it's because I made the craft wrong, or what.  GetSpoilerState works.  I have not tested SetSpoiler or Increase/Decrease Flaps

[RPM dev build](https://www.dropbox.com/s/2qf38gxg70sgbor/RasterPropMonitor-dev.0.21.1.zip?dl=0
), [FAR dev DLLs](https://www.dropbox.com/s/15josnjnu5f87jv/MOARdV-FAR-dev.zip?dl=0), and [MFI 1.1](http://forum.kerbalspaceprogram.com/threads/118088) are required.

The new methods (not documented on the wiki yet) are:

* JSIFAR:GetFlapSetting (returns 0-3, or -1 if it fails)
* JSIFAR:IncreaseFlapSetting (for JSIActionGroupSwitch)
* JSIFAR:IncreaseFlapSettingState (returns 1 if flaps can be increased)
* JSIFAR:DecreaseFlapSetting (for JSIActionGroupSwitch)
* JSIFAR:DecreaseFlapSettingState (returns 1 if flaps can be decreased)
* JSIFAR:SetSpoiler
* JSIFAR:GetSpoilerState (returns 1 if spoilers are deployed).

**Comment by MOARdV** [comment]
I've updated both the RPM dev build and the FAR dev DLLS.  I've also checked that all of the methods work, so you can start using that to develop flap and spoiler controls.  I don't know the FAR release timeline, but I can release v0.21.1 of RPM where it will have the methods in place so that the official FAR release will work, but the current FAR release won't cause harm.

**Comment by DeputyLOL** [comment]
thank you

**Comment by DeputyLOL** [comment]
I think a "set flaps position" would be useful.

**Comment by MOARdV** [comment]
I'll add in JSIFAR:SetFlaps0, JSIFAR:SetFlaps1, JSIFAR:SetFlaps2, and JSIFAR:SetFlaps3
